.github/workflows/deploy_tre_reusable.yml

```diff
@@ -180,6 +180,10 @@ jobs:
             echo "Missing variable: LOCATION" && exit 1
           fi
 
+          if [ "${{ secrets.CI_CACHE_ACR_NAME }}" == '' ]; then
+            echo "CI_CACHE_ACR_NAME is not set. Please run the workflow on the main branch first." && exit 1
+          fi
```

Comment:
Are you sure this was the problem you saw? i.e. this being an empty string?
Personally the issue I'd seen was that the login fails since the ACR doesn't exist as main didn't run.